### PR TITLE
Extract handle creation code into reusable method.

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Breaking out #6957 into smaller PRs.

### CODE REVIEW

File: javatests/arcs/core/host/ArcHostContextParticleTest.kt [hunk 1]
```diff
@@ -37,23 +36,18 @@ import org.junit.runners.JUnit4
...
+  fun setUp() {
+    schedulerProvider = SimpleSchedulerProvider(Dispatchers.Default)
     random = Random(System.currentTimeMillis())
-    particles = mutableListOf()
   }
 
   @After
   fun tearDown() = runBlocking {
     schedulerProvider.cancelAll()
+    RamDisk.clear()
```
Comment (mmandlis) [inline javatests/arcs/core/host/ArcHostContextParticleTest.kt:49]: did you revert the changes in this file on purpose, or did you not sync the latest changes?

Comment (alxrsngrtn) [reply javatests/arcs/core/host/ArcHostContextParticleTest.kt:49]: Ah, I may not have synced to the latest changes; I pulled this from another branch, which is likely stale.

Comment (alxrsngrtn) [reply javatests/arcs/core/host/ArcHostContextParticleTest.kt:49]: hmm... I pulled the changes with a `get checkout --` from another branch, so I'll need to do a manual revert on the relevant changes. Thanks for catching this.

Comment (alxrsngrtn) [reply javatests/arcs/core/host/ArcHostContextParticleTest.kt]: Should be fixed now.

**Review by mmandlis — APPROVED** [review]
